src/cace/amm/numeric.c

```diff
@@ -109,3 +110,165 @@ int cace_amm_numeric_promote_type(cace_ari_type_t *result, const cace_ari_t *lef
     }
     return 0;
 }
+
+int cace_numeric_integer_binary_operator(cace_ari_t *result, const cace_ari_t *lt_val, const cace_ari_t *rt_val,
+                                         cace_binop_uvast op_uvast, cace_binop_vast op_vast)
+{
```

Comment:
Any non-static function would be good to guard their inputs against null with `CHKERR1` since they are exported symbols and can be called from anywhere.